### PR TITLE
Store AKI in CertDB with Go 1.9 (#801)

### PR DESCRIPTION
**Pull request by mastersingh24**
Go 1.9 modified the x509.CreateCertificate
function to no longer update the AuthorityKeyId
on the template passed in.  cfssl signer relied
on this behavior in order to get the AKI and save
it in certdb.  This change now parses the signed
cert and uses the AKI from it.

### CODE REVIEW

File: signer/local/local.go [hunk 1]
```diff
@@ -387,12 +387,18 @@ func (s *Signer) Sign(req signer.SignRequest) (cert []byte, err error) {
 		return nil, err
 	}
 
+	// Get the AKI from signedCert.  This is required to support Go 1.9+.
+	// In prior versions of Go, x509.CreateCertificate updated the
+	// AuthorityKeyId of certTBS.
+	block, _ = pem.Decode([]byte(signedCert))
+	parsedCert, _ := x509.ParseCertificate(block.Bytes)
```
Comment (lziest) [inline signer/local/local.go]: you can do parsedCert, _ := helpers.ParseCertificatePEM(signedCert)
https://godoc.org/github.com/cloudflare/cfssl/helpers#ParseCertificatePEM

Comment (mastersingh24) [reply signer/local/local.go]: Cool.  Missed that.  Will modify

**Comment by mastersingh24** [comment]
What's the likelihood of this one getting merged?  My current project uses cfssl and we are moving to Go 1.9.  We can implement a workaround in our code, but we'd rather just vendor an updated version of cfssl with this fix in it

**Comment by kisom** [comment]
@mastersingh24 100% ;)

**Pull request merged**